pkg/kubelet/prober/prober_manager.go

```diff
@@ -84,7 +84,7 @@ type Manager interface {
 
 	// UpdatePodStatus modifies the given PodStatus with the appropriate Ready state for each
 	// container based on container running status, cached probe results and worker states.
-	UpdatePodStatus(types.UID, *v1.PodStatus)
+	UpdatePodStatus(types.UID, *kubecontainer.PodStatus, sets.Set[string])
```

Comment:
> And I think we need `container.RestartPolicy` field inside of this function as the ready state depends on the RestartPolicy value.

Yup that's correct.

> How about passing the pod directly?

I think it's a good point, although we need to clearly indicate the new flow kubecontainer -> v1 because it will become less clear from the signature.
Thoughts @SergeyKanzhelev ?

I will check later today if that helps with tests too...

Reply:
have it helped with tests?

Reply:
I have added the pod, passed as value to highlight it won't be changed by the method.
Tests from @smarterclayton, after a lot of tweaks back and forth, are correct and I need to correct transformations inside `generateAPIPodStatus`.
This is *not* as simple as it seemed ;_;